File: POS System/POS System/LoginInput.vb
Imports System
Imports System.Net
Imports System.Data.Sql
Imports System.Data.SqlClient
Imports System.Data.OleDb
Imports System.IO

Public Class LoginInput

    Dim currentpath As String = Directory.GetCurrentDirectory()


    Private Sub cmdLogin_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdLogin.Click

        UserName = txtUserName.Text.Trim
        UserPassword = txtPassword.Text.Trim
        GetConnectionString()

        If UserName = "admin" And UserPassword = "isha1234" Then
            SysOperator = "Admin"
            UserStatus = "Administrator"
            gblPrliviges = "1"
            UserID = "01"
            LoginStatus = True
            GoTo Login_Success
        End If

        Dim LoginQuery As String = "SELECT * FROM USERS WHERE USERNAME = N'" & UserName & "' AND PASSWORD = N'" & UserPassword & "'"
        ExecuteAdapterQuery(SqlDataAdapter, LoginQuery, Connection)
        Dim UsersLogin As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        If UsersLogin.HasRows = True Then
            While UsersLogin.Read
                gblPrliviges = UsersLogin.Item("SecurityLevel")
                UserStatus = UsersLogin.Item("UserRole")
                SysOperator = StrConv(UsersLogin.Item("Name"), VbStrConv.ProperCase)
                UserID = UsersLogin.Item("UserID")
                LoginStatus = True
                GoTo Login_Success
            End While
        Else
            MsgBox("Access Denied" & vbCrLf & "Invalid Passward.", MsgBoxStyle.Information, My.Application.Info.ProductName)
            Me.txtPassword.Clear()
            Connection.Close()
            Me.txtUserName.Focus()
        End If

Login_Success:

        If LoginStatus = True Then
            Main.Show()
            Connection.Close()
            Me.Close()
            Exit Sub
        End If

    End Sub

    Private Sub txtPassword_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles txtPassword.KeyDown
        If e.KeyCode = Keys.Enter Then
            If txtUserName.Text = "" Then MsgBox("UserName is blank.", MsgBoxStyle.Information) : Exit Sub
            cmdLogin_Click(sender, e)
        End If
    End Sub

    Private Sub txtUserName_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles txtUserName.KeyDown
        If e.KeyCode = Keys.Enter Then txtPassword.Focus()
    End Sub

    Private Sub cmdExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdExit.Click
        LoginStatus = False
        Me.Close()
    End Sub

    Private Sub frmLoginInput_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Copyright.Text = My.Application.Info.Copyright
        txtUserName.Focus()
    End Sub

End Class

File: POS System - Retaurant - 2022/POS System/LoginInput.cs
using System;
using System.IO;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace POS_System
{

    public partial class LoginInput
    {

        private string currentpath = Directory.GetCurrentDirectory();

        public LoginInput()
        {
            InitializeComponent();
        }


        private void cmdLogin_Click(object sender, EventArgs e)
        {

            mdlGeneral.UserName = txtUserName.Text.Trim();
            mdlGeneral.UserPassword = txtPassword.Text.Trim();
            mdlGeneral.GetConnectionString();
            // for developer only
            if (mdlGeneral.UserName == "admin" & mdlGeneral.UserPassword == "isha1234")
            {
                mdlGeneral.SysOperator = "Admin";
                mdlGeneral.UserStatus = "Administrator";
                mdlGeneral.gblPrliviges = Conversions.ToInteger("1");
                mdlGeneral.UserID = "01";
                mdlGeneral.LoginStatus = true;
                goto Login_Success;
            }

            string LoginQuery = "SELECT * FROM USERS WHERE USERNAME = N'" + mdlGeneral.UserName + "' AND PASSWORD = N'" + mdlGeneral.UserPassword + "'";
            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, LoginQuery, mdlGeneral.Connection);
            var UsersLogin = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            if (UsersLogin.HasRows == true)
            {
                while (UsersLogin.Read())
                {
                    mdlGeneral.gblPrliviges = Conversions.ToInteger(UsersLogin["SecurityLevel"]);
                    mdlGeneral.UserStatus = Conversions.ToString(UsersLogin["UserRole"]);
                    mdlGeneral.SysOperator = Strings.StrConv(Conversions.ToString(UsersLogin["Name"]), VbStrConv.ProperCase);
                    mdlGeneral.UserID = Conversions.ToString(UsersLogin["UserID"]);
                    mdlGeneral.LoginStatus = true;
                    goto Login_Success;
                }
            }
            else
            {
                Interaction.MsgBox("Access Denied" + Constants.vbCrLf + "Invalid Passward.", MsgBoxStyle.Information, My.MyProject.Application.Info.ProductName);
                txtPassword.Clear();
                mdlGeneral.Connection.Close();
                txtUserName.Focus();
            }

        Login_Success:
            ;


            if (mdlGeneral.LoginStatus == true)
            {
                My.MyProject.Forms.Main.Show();
                mdlGeneral.Connection.Close();
                Close();
                return;
            }

        }

        private void txtPassword_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                if (string.IsNullOrEmpty(txtUserName.Text))
                {
                    Interaction.MsgBox("UserName is blank.", MsgBoxStyle.Information);
                    return;
                }
                cmdLogin_Click(sender, e);
            }
        }

        private void txtUserName_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
                txtPassword.Focus();
        }

        private void cmdExit_Click(object sender, EventArgs e)
        {
            mdlGeneral.LoginStatus = false;
            Close();
        }

        private void frmLoginInput_Load(object sender, EventArgs e)
        {
            //FormBorderStyle = FormBorderStyle.None;
            Copyright.Text = My.MyProject.Application.Info.Copyright;
            txtUserName.Focus();
        }

    }
}